### PR TITLE
Adds the user to volunteer group

### LINKED ISSUE #64: Volunteering API
**Is your feature request related to a problem? Please describe.**
As user I should be able to volunteer to organisations

**Describe the solution you'd like**
1. Create api [get] /api/org/(org-id)/volunteer/ which adds the user sending the volunteer request into group "volunteer" of the organisation.
1. Make sure the user is not already a part of the org.
1. Add appropriate tests.
1. Add swagger docs for the same.

### PR DESCRIPTION
**Pull request by sahil9001**
### Description
Added the functionality to add the user to particular 'Volunteer' group of the organisation with a certain [org-id]. Not added tests as was having trouble writing it. 

**Type of Change:** (Delete irrelevant options)
- Code
- User Interface
- Documentation

**Code/Quality Assurance Only** (Delete irrelevant options)

- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
Tested locally without code, adds the user to the group.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by sahil9001** [comment]
@naveennvrgup sir I've added tests to the PR.

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]: Match the response according to swagger docs u wrote

expected:
```
{
    "detail": "Organization not present"
}
```

actual:
image

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
        volunteer_permission_set = PermissionSet.objects.get(
            name='Volunteer',
            org=org
        )
```
I think this line is redundant

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
        volunteer_group = Group.objects.get(
            name='Volunteer',
            org=org
        )
```

I think this line is redundant

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]: Match the response according to swagger docs u wrote

expected:
image

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]: 1. Please extract these responses to  `org.responses.py`
2. This response is not been documented 
image

File: tests/org/test_add_volunteer_api.py
Comment (naveennvrgup) [inline tests/org/test_add_volunteer_api.py:29]: Great job with the testcases I would like you to add:
1. test_fail_bad_org_id
1. test_fail_already_member_of_org

**Review by naveennvrgup — CHANGES_REQUESTED** [review]
1. Great job with API :clap: 
1. Please address my comments.

**Comment by sahil9001** [comment]
@naveennvrgup I've modified the files as per you asked, please review it once again.

File: org/views.py
Comment (naveennvrgup) [inline org/views.py]:
```suggestion
        '409': set_example(responses.user_already_present_200)
```
 I guess "409 conflict" will be more appropriate :thinking: . What u think?

File: tests/org/test_add_volunteer_api.py
Comment (naveennvrgup) [inline tests/org/test_add_volunteer_api.py]: A better way to do the setup will be to call the create org API or use the CreateOrgSerializer which has a custom save() has present.

File: tests/org/test_add_volunteer_api.py
Comment (naveennvrgup) [inline tests/org/test_add_volunteer_api.py]: By using `tests.AuthAPITestCase.AuthAPITestCase` we can avoid this code duplication. 

Ex. `tests.org. CreateOrgAPITestCase`

File: tests/org/test_add_volunteer_api.py
Comment (naveennvrgup) [inline tests/org/test_add_volunteer_api.py]: Code duplication in this function also. Try to use `AuthAPITestCase`

**Review by naveennvrgup — CHANGES_REQUESTED** [review]
1. The API looks almost perfect
1. The tests logic is fine but implementation will need a redo :muscle: 
1. Great work so far!

File: tests/org/test_add_volunteer_api.py
Comment (sahil9001) [reply tests/org/test_add_volunteer_api.py]: I'm having trouble setting this can't we use the existing code for set up?

File: tests/AuthAPITestCase.py
Comment (naveennvrgup) [inline tests/AuthAPITestCase.py]: 1. I suppose you misunderstood the suggestions about `tests/AuthAPITestCase.py `.
1. The purpose is not to alter this file but to use it ( instead of duplicating the `create_auth_client()` in every test.
1. Please revoke the changes you made to this file :smiley:

File: tests/org/test_add_volunteer_api.py
Comment (naveennvrgup) [reply tests/org/test_add_volunteer_api.py]: I will provide you better explanation over call

**Comment by sahil9001** [comment]
@naveennvrgup sir I've made the following changes:

1. Removed duplicated code 
2. Modified test_already_member function to make request 2 times.
3. Used CreateOrgSerializer() to create org and inherited setUp() from AuthAPITestCase

**Review by naveennvrgup — APPROVED** [review]
1. Fantastic work :fire: 
1. Works as expected, tested locally.
1. Can be merged after 2 approval :+1:

**Review by priyanshisharma — APPROVED** [review]
Code looks good to me! Great job @sahil9001 🔥

**Pull request merged**
